### PR TITLE
node: add ReadStream and createReadStream

### PR DESCRIPTION
**Pull request by bartlomieju**
Mocking some more modules to try `vite` working in `--compat` mode.

### CODE REVIEW

File: node/child_process.ts [hunk 1]
```diff
@@ -318,7 +318,21 @@ export function spawn(
...
+  argsOrOptions?: string[] | SpawnOptions,
+  maybeOptions?: SpawnOptions,
+): ChildProcess {
+  const args = Array.isArray(argsOrOptions) ? argsOrOptions : [];
+  const options = !Array.isArray(argsOrOptions) && argsOrOptions != null
+  ? argsOrOptions
+  : maybeOptions;
+  args.push(modulePath)
+  console.log("fork", Deno.execPath, args, options);
+  return new ChildProcess(Deno.execPath(), args, options);
```
Comment (bartlomieju) [inline node/child_process.ts:332]: This is wrong, it should actually imply `deno run --compat --unstable --allow-read`

File: node/module.ts [hunk 2]
```diff
@@ -757,6 +757,19 @@ nativeModulePolyfill.set(
...
+    wrap: Module.wrap,
+    _initPaths: Module._initPaths,
+    _preloadModules: Module._preloadModules,
+    _resolveLookupPaths: Module._resolveLookupPaths,
+    _cache: Module._cache,
+    _pathCache: Module._pathCache,
+    _extensions: Module._extensions,
+    globalPaths: Module.globalPaths,
+    _findPath: Module._findPath,
+    Module,
```
Comment (bartlomieju) [inline node/module.ts:772]: Open a separate PR for this

File: node/process.ts [hunk 3]
```diff
@@ -17,13 +18,13 @@ const notImplementedEvents = [
...
-  "uncaughtException",
+  // "uncaughtException",
   "uncaughtExceptionMonitor",
-  "unhandledRejection",
+  // "unhandledRejection",
 ];
 
 /** https://nodejs.org/api/process.html#process_process_arch */
-export const arch = Deno.build.arch;
+export const arch = osArch();
```
Comment (bartlomieju) [inline node/process.ts:27]: Open a separate PR for this

File: node/process.ts [hunk 4]
```diff
@@ -345,6 +349,8 @@ class Process extends EventEmitter {
    */
   argv = argv;
 
+  execArgv = [];
```
Comment (bartlomieju) [inline node/process.ts:352]: Open a separate PR for this

File: node/module_all.ts [hunk 5]
```diff
@@ -69,7 +75,13 @@ export default {
   timers,
   "timers/promises": timersPromises,
   tty,
+  tls: {},
   url,
+  "worker_threads": {},
   util,
+  vm: {
+    Script,
+  },
```
Comment (bartlomieju) [inline node/module_all.ts:84]: Open a separate PR for this and make methods on `Script` throw not-implemented error

File: node/_fs/_fs_unlink.ts [hunk 6]
```diff
@@ -4,5 +4,12 @@ export function unlink(path: string | URL, callback: (err?: Error) => void) {
...
 export function unlinkSync(path: string | URL) {
-  Deno.removeSync(path);
+  try {
+    Deno.removeSync(path);
+  } catch (e) {
+    if (e instanceof Deno.errors.NotFound) {
+      e.code = "ENOENT";
+    }
+    throw e;
+  }
```
Comment (bartlomieju) [inline node/_fs/_fs_unlink.ts:14]: This is the last bit needed to get ESLint working

Comment (bartlomieju) [reply node/_fs/_fs_unlink.ts:14]: Ref https://github.com/denoland/deno/pull/12591

Comment (AaronO) [reply node/_fs/_fs_unlink.ts:14]: @bartlomieju Can you try the latest version of https://github.com/denoland/deno/pull/12591 ?

Comment (bartlomieju) [reply node/_fs/_fs_unlink.ts:14]: It works with that PR

File: node/process.ts [hunk 7]
```diff
@@ -467,6 +470,10 @@ class Process extends EventEmitter {
     const [prevSec, prevNano] = time;
     return [sec - prevSec, nano - prevNano];
   }
+  
+  memoryUsage() {
+    return Deno.memoryUsage();
+  }
```
Comment (bartlomieju) [inline node/process.ts:476]: @AaronO you probably want to add `arrayBuffers` with the value 0

Comment (AaronO) [reply node/process.ts:476]: Yep, will do. I had suggested adding that field in the original Deno PR

**Review by kt3k — APPROVED** [review]
Nice! LGTM

**Pull request merged**